### DISCUSSION TITLE
Replace melody-* package with twig-lexer

### DISCUSSION THREAD

**Discussion by zackad** [zackad/prettier-plugin-twig#28] [Ideas]
@rellafella mention about [twig-lexer](https://www.npmjs.com/package/twig-lexer) that might be able to replace `melody-*` dependencies. If this could be implemented it might reduce complexity of the project and we no longer need to integrate `melody-*` package into this project directly.

Related:
- #26

**Comment by zackad** [comment]
What we need is a parser, lexer is part of the parsing process. We still need to build AST ourself (assuming we drop `melody-*` packages). I think I'll go with original plan to merge `melody-*` packages directly into this project and gradually offload tokenizing process into external library if that will make it easier to manage.

**Reply by rellafella** [reply]
I haven't had a good amount of time to have a decent look at this. I think lets try and bundle those melody-* packages in and at least we are moving towards less unmaintained dependencies and hopefully can get a clearer picture into what is required.

**Comment by rellafella** [comment]
So it turns out a lot of the melody stuff, even though saved as .js files is actually written in typescript. Even through it would be great to have the whole repo in typescript I think the best way ahead would be to convert what we can to JSDoc as we integrate the packages into the core of this plugin.
